extensions/kubernetes/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesConfig.java

```diff
@@ -0,0 +1,24 @@
+
+package io.quarkus.kubernetes.deployment;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+@ConfigRoot
+public class KubernetesConfig {
+
+    private static final String KUBERNETES = "kubernetes";
+    private static final String OPENSHIFT = "openshift";
+    private static final String KNATIVE = "knative";
```

Comment:
Can we have an enum `DeploymentTarget` for these values? 

This way, accepted values will be probably displayed in the generated docs.

Reply:
+1 on that one.

Reply:
Good to know about the docs

Reply:
Done